test/models/vote_test.rb

```diff
@@ -0,0 +1,66 @@
+require "test_helper"
+
+describe Vote do
+
+  before do
+    @work = Work.create!(category: "album", title: "melodrama", creator: "lorde", publication_year: 2017, description: "best albums from Lorde")
+    @user = users(:rabbit)
+  end
+
+  describe "validations" do
+
+    it "can be created with all fields" do
+
+      vote1 = Vote.new(user_id: @user.id, work_id: @work.id)
+      result = vote1.valid?
+      result.must_equal true
+    end
+
+    it "requires a work id" do
+      vote = Vote.new(user_id: @user)
+      result = vote.valid?
+      result.must_equal false
+
+      vote.errors.messages.must_include :work_id
+    end
+
+    it "requires a user id" do
+
+      vote = Vote.new(work_id: @work)
+      result = vote.valid?
+      result.must_equal false
+
+      vote.errors.messages.must_include :user_id
+    end
+
+    it "don't allow user to vote more than once for one work" do
+      user = User.new(name: "yellowlion")
+      vote = Vote.new(user: user, work: @work)
+      vote2 = Vote.new(user: user, work: @work)
+      result = vote2.valid?
```

Comment:
This test works, because `vote2` is not valid because the user doesn't have an id, not because it's duplicating a vote (because `vote` was not saved).

You should instead do:

```ruby
user = User.create(name: "yellowlion")
original_vote = Vote.create(user: user, work: @work) # notice I'm saving here
duplicate_vote = Vote.new(user: user, work: @work)

duplicate_vote.valid?.must_equal false
```